web/src/click_handlers.js

```diff
@@ -399,7 +400,26 @@ export function initialize() {
         const topic_name = $(e.target).attr("data-topic-name");
         message_edit.toggle_resolve_topic(message_id, topic_name);
     });
-
+    document.body.addEventListener("click", (e) => {
+        if (e.target.matches(".on_hover_topic_resolve")) {
+            e.stopPropagation();
+            const element = e.target; // Get the clicked element
+            const topic_name = $(e.target).attr("data-topic-name");
+            // Update class to show the spinner
+            element.classList.remove("fa-check");
+            element.classList.add("fa-spinner");
+            const timer = setInterval(() => {
+                // If the request is still ongoing, keep the icon as a spinner
+                // The request is complete, so change the icon back to a check mark
+                if (!resolved_topic.is_resolved(topic_name)) {
+                    return;
+                }
+                element.classList.remove("fa-spinner");
+                element.classList.add("fa-check");
+                clearInterval(timer);
+            }, 1000);
```

Comment:
Please don't use setInterval polling like this -- we get to run code when the request completes, and hooks to change this state should use those.

I recommend studying other spinners we have, like when saving a change in the "move messages" modal, for better patterns.